examples/authorized_api/authorization.go

```diff
@@ -3,10 +3,59 @@ package main
 import (
 	"encoding/json"
 	"errors"
+	"fmt"
 	"io/ioutil"
 	"net/http"
 )
 
+func ZebedeeMustPermitUser(handler http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		status, err := zebedeeValidateUser()
+		if err != nil {
+			if status == "401 Unauthorized" {
+				http.Error(w, err.Error(), http.StatusUnauthorized)
+			} else {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+			}
+			return
+		} else {
+			w.WriteHeader(200)
+			message := "[\"DELETE\", \"READ\", \"CREATE\", \"UPDATE\"]"
+			fmt.Fprintf(w, message)
+		}
+		handler(w, r)
+	}
```

Comment:
The `else` block isn't necessary here you can do:
```
return func(w http.ResponseWriter, r *http.Request) {
		status, err := zebedeeValidateUser()
		if err != nil {
			if status == "401 Unauthorized" {
				http.Error(w, err.Error(), http.StatusUnauthorized)
			} else {
				http.Error(w, err.Error(), http.StatusInternalServerError)
			}
			return
		} 
		
		w.WriteHeader(200)
		message := "[\"DELETE\", \"READ\", \"CREATE\", \"UPDATE\"]"
		fmt.Fprintf(w, message)
		handler(w, r)
	}
```

Reply:
yes I prefer that style - changed now